### ISSUE TITLE
Single file mode tests

### ISSUE THREAD

**Issue by manodasanW** [microsoft/CsWinRT#999]
To the least, we should have a test which publishes a console app consuming a C#/WinRT projection to single file and run it in our test pass to confirm that it runs without issues.

**Comment by kant2002** [comment]
Probably collection of small applications which tests specific combinations of marshalling would be proper way to test single-file mode. I thinking about similar approach as .NET runtime has, where test is executable whcih runs and produce well known non-zer exit code to indicate correct pass. Same projects can be used to exercise regular publish process and single-file mode. Also I envision that it would be easy to retrofit such approach with NativeAOT flavor if would be a need for this. Can write more specific

**Comment by kant2002** [comment]
Please can somebody from the team help me setup test for single-file scenario. I'm a bit lost in the woods what's minimal amount of work required.

**Comment by manodasanW** [comment]
@kant2002 I am hoping the basic functional tests I am working on right now as part of changes for supporting IL trimming will also help serve as tests for validating single file publishing.

**Comment by kant2002** [comment]
Then I just hope you can land https://github.com/microsoft/CsWinRT/pull/993 later one. Can you give link to your PR? I would like to subscribe.

**Comment by kant2002** [comment]
@manodasanW I notice that IL trimming work starts flowing, so curios what's needed for closing this issue. Honestly #993 is quite aged, and because I need it for much bigger work (potentially) I'm a bit frustrated about how long did it take to land small change like this.

Can I help somehow? or maybe #993 may lands without this one (assuming CsWinRT do not consider NativeAOT supported scenario). I worrying that internally for you make sense combine this together with #1141 which indicates that work not even started, but for NativeAOT I and couple other individuals can work relatively independently try to find what's issues may arise.

**Comment by manodasanW** [comment]
@kant2002 I agree that this took longer than we expected to get to.  But I finally got my PR out for my first set of IL trimming changes (https://github.com/microsoft/CsWinRT/pull/1224) which also comes with a set of functional tests that should satisfy this issue.  Once that PR is merged, I will have a follow up to enable SingleFile in the publish profile which I have confirmed fails without your change and as part of that, we can also merge your PR.
